src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java

```diff
@@ -337,8 +337,7 @@ private boolean areExternalResourcesChanged(Set<ExternalResource> resources) {
      * @param externalResources a set of {@link ExternalResource}.
      */
     private void fillCacheWithExternalResources(Set<ExternalResource> externalResources) {
-        externalResources.stream()
-            .filter(resource -> !isResourceLocationInCache(resource.location))
+        externalResources
             .forEach(resource -> details.setProperty(resource.location, resource.contentHashSum));
```

Comment:
This is only called after `areExternalResourcesChanged` method and `reset`. So at this point we already know all resources are not in the cache and must populate all of them. Since this is called after a reset, we always re-populate all resources and so check is not needed.

Maybe list of resources can have the same file location twice, but then there is no harm in populating it twice. If it was an issue, then maybe the set should be removing the duplicates. It isn't now because ExternalResource is the Check and they don't define hashCode/compareTo.

Reply:
confirmed.